contrib/cindercompatibleapi/converter/volumetype.go

```diff
@@ -0,0 +1,342 @@
+	AccessIsPublic bool      `json:"os-volume-type-access:is_public,omitempty"`
+	Description    string    `json:"description,omitempty"`
+	Extras         ExtraSpec `json:"extra_specs,omitempty"`
+}
+
+// CreateTypeRespSpec ...
+type CreateTypeRespSpec struct {
+	VolumeType CreateRespVolumeType `json:"volume_type,omitempty"`
+}
+
+// CreateRespVolumeType ...
+type CreateRespVolumeType struct {
+	IsPublic       bool      `json:"is_public,omitempty"`
+	Extras         ExtraSpec `json:"extra_specs,omitempty"`
+	Description    string    `json:"description,omitempty"`
+	Name           string    `json:"name,omitempty"`
+	ID             string    `json:"id,omitempty"`
+	AccessIsPublic bool      `json:"os-volume-type-access:is_public,omitempty"`
+}
+
+// CreateTypeReq ...
+func CreateTypeReq(cinderReq *CreateTypeReqSpec) (*model.ProfileSpec, error) {
+	profile := model.ProfileSpec{}
+
+	profile.Name = cinderReq.VolumeType.Name
+	if false == cinderReq.VolumeType.AccessIsPublic {
+		return nil, errors.New("OpenSDS does not support os-volume-type-access:is_public = false")
+	}
+	profile.Description = cinderReq.VolumeType.Description
+	profile.Extras = *(CinderExtraToOpenSDSExtra(&(cinderReq.VolumeType.Extras)))
+
+	// The storageType can be block, file, object, default is block
+	profile.StorageType = "block"
+
+	return &profile, nil
+}
+
+// CinderExtraToOpenSDSExtra ...
+func CinderExtraToOpenSDSExtra(typeExtra *ExtraSpec) *model.ExtraSpec {
+	var profileExtras model.ExtraSpec
+	profileExtras = make(map[string]interface{})
+	for key, value := range *typeExtra {
+		profileExtras[key] = value
+	}
+
+	return &profileExtras
+}
+
+// CreateTypeResp ...
+func CreateTypeResp(profile *model.ProfileSpec) *CreateTypeRespSpec {
+	resp := CreateTypeRespSpec{}
+	resp.VolumeType.IsPublic = true
+	resp.VolumeType.Extras = *(OpenSDSExtraToCinderExtra(&(profile.Extras)))
+	resp.VolumeType.Description = profile.Description
+	resp.VolumeType.Name = profile.Name
+	resp.VolumeType.ID = profile.BaseModel.Id
+	resp.VolumeType.AccessIsPublic = true
+
+	return &resp
+}
+
+// *******************Update a volume type*******************
+
+// UpdateTypeReqSpec ...
+type UpdateTypeReqSpec struct {
+	VolumeType UpdateReqVolumeType `json:"volume_type"`
+}
+
+// UpdateReqVolumeType ...
+type UpdateReqVolumeType struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	IsPublic    bool   `json:"is_public,omitempty"`
+}
+
+// UpdateTypeRespSpec ...
+type UpdateTypeRespSpec struct {
+	VolumeType UpdateRespVolumeType `json:"volume_type"`
+}
+
+// UpdateRespVolumeType ...
+type UpdateRespVolumeType struct {
+	IsPublic    bool      `json:"is_public"`
+	Extras      ExtraSpec `json:"extra_specs"`
+	Description string    `json:"description"`
+	Name        string    `json:"name"`
+	ID          string    `json:"id"`
+}
+
+// UpdateTypeReq ...
+func UpdateTypeReq(cinderReq *UpdateTypeReqSpec) (*model.ProfileSpec, error) {
```

Comment:
I don't think this method should return error, since other methods only return ***RespSpec. You can package the validate operation in a standalone method.